File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>>map;
        vector<vector<string>> res;
        string temp;

        for(int i = 0 ; i < strs.size(); i++){
            temp = strs[i];
            sort(begin(temp), end(temp));
            map[temp].push_back(strs[i]);
        }

        for(auto& p : map){
            res.push_back(p.second);
        }

        return res;
    }
};

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        freq = defaultdict(list)

        for s in strs:
            count = [0]*26
            for c in s:
                count[ord(c) - ord('a')] += 1
            keys = tuple(count)
            freq[keys].append(s)
        
        return list(freq.values())